lib/util/Strings.v3

```diff
@@ -133,16 +133,22 @@ component Strings {
 		if (pos >= max) return (0, null);
 		if (a[pos] != '\"') return (0, null);
 		var i = pos + 1, buf = StringBuilder.new();
+		var checked = false;
 		while (i < max) {
 			var ch = a[i];
-			if (ch < ' ' || ch > 127) return (pos - i, null);
+			if (ch < ' ') return (pos - i, null);
 			if (ch == '\"') return (1 + i - pos, buf.toString());
 			if (ch == '\\') {
 				var p = Chars.parseEscape(a, i + 1);
 				if (p.0 <= 0) return (pos - i + p.0, null);
 				else buf.putc(p.1);
 				i = i + p.0;
 			} else {
+				if (ch > 127 && !checked) {
```

Comment:
As I understand this code, upon the first non-ascii character, this checks the rest of the input buffer (i.e. until the end, which could be past the closing `"`) and then proceeds to just add the non-ascii byte to the buffer. It only does this once, but I still think that's not quite right, because after adding the first byte of the UTF-8 encoded character, it will just go back through the loop with the next byte. I think this should probably decode the whole codepoint and add those bytes to the buffer, then continue in the loop.

Reply:
OK. I'll fix it. I wrongly assume that `max` is the end of the string.